### ISSUE TITLE
Add a command for querying std doc

### ISSUE THREAD

**Issue by upsuper** [upsuper/telegram-rustevalbot#19]
Maybe `/doc` and return the first result which is fully matched.

**Comment by upsuper** [comment]
It seems rustdoc uses an index file for that. We would need to parse https://doc.rust-lang.org/search-index.js somehow.

**Comment by blackgear** [comment]
https://play.rust-lang.org/?gist=aac074e82d5305cf4a5b562013470a71&version=stable&mode=debug&edition=2015

Here is a simple parser for search-index...

**Comment by blackgear** [comment]
I'd like to write a mod like this:

```rust
pub struct RustDoc {
    items: Vec<IndexItem>,
    index: fst::Map,
};

impl RustDoc {
    pub fn new(data: &str) -> RustDoc {};
    pub fn search<'a>(&'a self, keyword: &str) -> Vec<&'a IndexItem> {};
}
```

`fst::Map` stores key-value map: `<[u8], usize>`, the `usize` indicate the index of `Vec<IndexItem>`.

I'd like to use `"{path}/{name} {desc}"` as key, then fuzz search with `fst_levenshtein`, using edit-distance.

Maybe a `fst::Map` should be separated depends on index build time?

**Comment by blackgear** [comment]
https://crates.io/crates/rustdoc-seeker

Ver 0.1.0 released as an technology preview, let's decide the search query format and result format~

**Comment by blackgear** [comment]
Ver 0.3.0 should have a stable api..

Edit distance with 30% err rate combined with starts_with looks solid:
```
let distance = (keyword.len() as f32 * 0.3) as u32;
let aut = fst_levenshtein::Levenshtein::new(keyword, distance).unwrap().starts_with();
for i in seeker.search(&aut) {...}
```

fst::automaton::Subsequence looks fine to me, but "eq" will matching "exxxxxq" causing confusing...

**Comment by upsuper** [comment]
Implemented in 5f88ff8a1cd2d7d86a0876a017290a871a119828 .. 594813f69b1412bfed136b03cac6ef0f8f2fb2fc. Follow up in #29.
